docs/develop/java/versioning.mdx

```diff
@@ -206,7 +163,7 @@ private final MyActivity activity =
         MyActivity.class,
         ActivityOptions.newBuilder()
           .setScheduleToCloseTimeout(Duration.ofSeconds(10))
-          .setVersioningIntent(VersioningIntent.VERSIONING_INTENT_DEFAULT)
+          .setVersioningIntent(VersioningIntent.VERSIONING_INTENT_USE_ASSIGNMENT_RULES)
```

Comment:
This is not accurate yet, but will be soon.  But it's quite challenging to document the current state, so I just stuck an "under construction" caution callout just above.